testing/test_script.py

```diff
@@ -1,18 +1,250 @@
+    initial_config = json.load(file_conf)
+    file_conf.close()
+    file_conf = open("testing_data/model_config_auto.json", "w")
+    initial_config["command"] = "train"
+    initial_config["gpu"] = "[7]"
+    json.dump(initial_config, file_conf)
+
+
+def test_automate_training():
+    command = "ivadomed_automate_training -c testing_data/model_config_auto.json " \
+              "-p testing_data/hyperparameter_opt.json -n 1 --fixed-split"
+    subprocess.check_output(command, shell=True)
+
+    command = "ivadomed_automate_training -c testing_data/model_config_auto.json " \
+              "-p testing_data/hyperparameter_opt.json -n 1 --run-test --all-combin"
+    subprocess.check_output(command, shell=True)
+
+
+def test_create_json_3d_unet_test():
+    # modify train config
+    null = None
+    command = "cp ivadomed/config/config_tumorSeg.json testing_data/model_config_3d.json"
+    subprocess.check_output(command, shell=True)
+    file_conf = open("testing_data/model_config_3d.json", "r")
+    initial_config = json.load(file_conf)
+    file_conf.close()
+    file_conf = open("testing_data/model_config_3d.json", "w")
+    initial_config["command"] = "test"
+    initial_config["loader_parameters"] = {
+        "target_suffix": ["_lesion-manual"],
+        "roi_suffix": null,
+        "bids_path": "testing_data",
+        "roi_params": {
+            "suffix": null,
+            "slice_filter_roi": null
+        },
+        "contrast_params": {
+            "training_validation": ["T2w"],
+            "testing": ["T2w"],
+            "balance": {}
+        },
+        "slice_filter_params": {
+            "filter_empty_mask": False,
+            "filter_empty_input": True
+        },
+        "slice_axis": "sagittal",
+        "multichannel": False,
+        "soft_gt": False
+    }
+    initial_config["log_directory"] = "3d_test"
+    initial_config["UNet3D"] = {
+        "applied": True,
+        "length_3D": [48, 48, 16],
+        "stride_3D": [48, 48, 16],
+        "attention": False,
+        "n_filters": 8
+    }
+    initial_config["transformation"] = {
+        "Resample":
+            {
+                "wspace": 1,
+                "hspace": 1,
+                "dspace": 1,
+                "preprocessing": True
+            },
+        "CenterCrop": {"size": [48, 48, 16], "preprocessing": True},
+        "NumpyToTensor": {},
+        "NormalizeInstance": {"applied_to": ["im"]}
+    }
+    json.dump(initial_config, file_conf)
+
+
+def test_create_model_unet3d():
+    model = imed_models.UNet3D(in_channel=1, out_channel=1, n_filters=8)
+    torch.save(model, "model_unet_3d_test.pt")
+    os.makedirs("3d_test")
+    command = "cp model_unet_3d_test.pt 3d_test/best_model.pt"
+    subprocess.check_output(command, shell=True)
+
+
+def test_testing_unet3d():
+    subprocess.check_output(["ivadomed -c testing_data/model_config_3d.json"], shell=True)
+
+
+def append_list_as_row(file_name, list_of_elem):
```

Comment:
Idem

Reply:
This is to add row to the csv file (particpant.csv)

Reply:
But it wouldn't be necessary if all the data was in ivadomed/data-testing

Reply:
Yes but then you will have a15 mb data-testing that juste stores identical images.

Reply:
I don't see why this is a problem, this repo exists for this exact purpose, no?

Reply:
i would still encourage to keep this repos at a minimum size (it is downloaded very often by CI and local developers)

Reply:
We are talking about 4 MB extra, but if we wanted to keep this dataset to a minimal size we could use smaller images or dummy images (`sub-unf01_T2star.nii.gz` and `sub-unf01_T2star.nii.gz` weight about 1 MB each which represents pretty much all the dataset's weight). I just feel like doing a lot of extra code to create more data defies the purpose to have a testing dataset that should include all the data we need to test, plus it is not very elegant.  Cropping the images around the spinal cord would limit the size of the dataset and would allow us to have more test subjects.

Reply:
Data were already cropped (eventhough it is not just around the spinal cord)

Reply:
ok i need a bit of context here-- based on the discussion thread, i understand that the discussion is about duplicating some of the testing data in order to test what happens if ivadomed receives 2 subjects instead of 1, is that correct?

if this is indeed correct, then this would justify doing the copy on the fly by the testing code (ie not duplicating the data)

Reply:
This is the context: 
for training a model with ivadomed, right now, you need to have a training set, a validation set, and a test set. We load images per subject. So If you want three sets, you will need to have three patients. 
For now, during the testing phase, I duplicate what I have for one subject (derivatives, images, metadata) to create 2 "new" ones. This allows me to run the full ivadomed workflow (integration tests) with 1 subject as the training set, 1 as the validation and 1 as the testing set.

Reply:
ok got it-- i'm not sure what's the best strategy here. I am leaning towards adding 2 other subjects in the testing-data repository, but with extreme cropping (to be greedy on the repos size)

Reply:
I think the main issue is that right now it is kind of an hybrid approach. We can either add subject or go with your suggestion from #392 and create everything (since we don't need to much anatomical meaning in the image considering the training is just 1 epochs we don't test performance)